### PR TITLE
[CALCITE-5783] Make the TableFunctionScan hintable

### PR DESCRIPTION
**Pull request by Aitozi**
This PR is meant to make the `TableFunctionScan` hintable

### CODE REVIEW

File: core/src/test/java/org/apache/calcite/test/SqlHintsConverterTest.java [hunk 1]
```diff
@@ -931,6 +937,12 @@ private static class HintCollector extends RelShuttleImpl {
           if (snapshot.getHints().size() > 0) {
             this.hintsCollect.add("Snapshot:" + snapshot.getHints());
           }
+        } else if (other instanceof LogicalTableFunctionScan) {
```
Comment (libenchao) [inline core/src/test/java/org/apache/calcite/test/SqlHintsConverterTest.java:940]: Prefer to use the base class `TableFunctionScan`.

**Review by libenchao — COMMENTED** [review]
@Aitozi Thanks for your PR, it looks great, I only have one minor comment.

File: core/src/test/java/org/apache/calcite/test/SqlHintsConverterTest.java (hunk 1 above)
Comment (Aitozi) [reply core/src/test/java/org/apache/calcite/test/SqlHintsConverterTest.java]: Updated

**Review by libenchao — APPROVED** [review]
+1

**Pull request closed without merging**